receiver/podmanreceiver/factory.go

```diff
@@ -52,17 +53,53 @@ func createDefaultReceiverConfig() config.Receiver {
 	return createDefaultConfig()
 }
 
+func createReceiver(
+	ctx context.Context,
+	params component.ReceiverCreateSettings,
+	config config.Receiver,
+) (*receiver, error) {
+	podmanConfig := config.(*Config)
+	var err error
+	r := receivers[podmanConfig]
+	if r == nil {
+		r, err = newReceiver(ctx, params, podmanConfig, nil)
+		if err != nil {
+			return nil, err
+		}
+		receivers[podmanConfig] = r
```

Comment:
Why we have to share the receiver instance? If it's really needed please handle concurrent access to the map

Reply:
We do not want to make different structures for metrics receiver and logs receiver as we are able to do it in a single structure but we have handled concurrent access to the map.